### PR TITLE
[MNG-8117] Maven PluginVersionResolver improvement

### PR DESCRIPTION
**Pull request by cstamas**
Produces output like this:
https://gist.github.com/cstamas/3ca4bc6cea5f701054061871b5db3f35

---

https://issues.apache.org/jira/browse/MNG-8117

### CODE REVIEW

**Review by kwin — APPROVED** [review]
LGTM, thanks

File: maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java [hunk 1]
```diff
@@ -261,8 +291,12 @@ private boolean isCompatible(PluginVersionRequest request, String version) {
 
         try {
             pluginManager.checkPrerequisites(pluginDescriptor);
-        } catch (Exception e) {
-            logger.warn("Ignoring incompatible plugin version {}", version, e);
+        } catch (PluginIncompatibleException e) {
+            if (logger.isDebugEnabled()) {
+                logger.warn("Ignoring incompatible plugin version {}: {}", version, e.getMessage(), e);
```
Comment (michael-o) [inline maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java:296]: This looks duplicate when you pass `e` already.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java (hunk 1 above)
Comment (cstamas) [reply maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java:296]: Agreed, all I wanted to distinguish is "with stack trace" or "without stack trace".

Comment (michael-o) [reply maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java:296]: I know, but that would rather confuse when the message appears twice

Comment (cstamas) [reply maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java:296]: Well, mvn -X output is already quite confusing :smile: so unsure will _this_ confuse anyone... but agreed, will fix.

Comment (cstamas) [reply maven-core/src/main/java/org/apache/maven/plugin/version/internal/DefaultPluginVersionResolver.java]: done

**Review by michael-o — APPROVED** [review]
No objections

**Comment by jira-importer** [comment]
Resolve #9990

**Pull request merged**